ethpm_cli/scraper.py

```diff
@@ -0,0 +1,195 @@
+from collections import OrderedDict
+import itertools
+import json
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Set, Tuple  # noqa: F401
+
+from eth_utils import to_dict, to_hex, to_list
+from eth_utils.toolz import assoc
+from ethpm.typing import URI, Address
+from ethpm.utils.backend import resolve_uri_contents
+from ethpm.utils.ipfs import extract_ipfs_path_from_uri, is_ipfs_uri
+from ethpm.utils.uri import is_supported_content_addressed_uri
+from web3 import Web3
+from web3._utils.events import get_event_data
+
+from ethpm_cli._utils.various import flatten
+from ethpm_cli.constants import (
+    CONTENT_DIR,
+    EVENT_ABI as VERSION_RELEASE_EVENT_ABI,
+    TOPIC,
+)
+from ethpm_cli.exceptions import BlockNotFoundError, InstallError
+from loggin import Logger
+
+
+class Scraper:
+    """
+    Scrapes blockchain for all valid 'VersionRelease' events
+    Resolves manifest uris & nested uris to a json object
+    """
+
+    def __init__(
+        self,
+        w3: Web3,
+        ws_w3: Web3 = None,
+        content_dir: Path = None,
+        logger: Logger = None,
+    ) -> None:
+        self.w3 = w3
+        # ws w3 instance needed since infura requires ws for certain jsonrpc calls
```

Comment:
@kclowes - Infura only supports [certain jsonrpc](https://infura.io/docs) endpoints over ws (i.e. `eth_newFilter`) and others only over https (i.e. `eth_getBlock`).  I had a thought that it might be worth making web3 smart enough to direct the right calls to the relevant endpoints - and avoid having to use 2 different w3 instances - but it only applies to infura, so I'm not sure if it's worth it, just a thought.

Reply:
@kclowes scratch that re: [this](https://github.com/ethpm/ethpm-cli/pull/3#discussion_r279393654) `local_filter_middleware` works great!